File: problem-set-1/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

int get_height(void);
void print_pyramids(int height, int width);
void repeat(string symbol, int times);

int main(void)
{
    int height = get_height();
    int width = height;

    print_pyramids(height, width);
}

int get_height(void)
{
    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    return height;
}

void print_pyramids(int height, int width)
{
    for (int i = 1; i <= height; i++)
    {
        int hash_count = i;
        int space_count = width - hash_count;

        repeat(" ", space_count);
        repeat("#", hash_count);

        printf("  ");

        repeat("#", hash_count);

        printf("\n");
    }
}

void repeat(string symbol, int times)
{
    for (int i = 0; i < times; i++)
    {
        printf("%s", symbol);
    }
}

File: introduction-to-computer-science/week-6/mario/more/mario.py
def main():
    height = width = get_height()
    print_pyramids(height, width)


def get_height():
    while True:
        try:
            height = int(input("Height: "))
            if height > 0 and height < 9:
                break
        except ValueError:
            None

    return height


def print_pyramids(height, width):
    for i in range(1, height + 1):
        hash_count = i
        space_count = width - hash_count

        print(" " * space_count, end="")
        print("#" * hash_count, end="")

        print("  ", end="")

        print("#" * hash_count)


main()
